client/state/themes/selectors/get-theme-signup-url.js

```diff
@@ -19,19 +20,41 @@ export function getThemeSignupUrl( state, themeId, options = {} ) {
 
 	const { tabFilter, styleVariationSlug } = options;
 	const themeType = getThemeType( state, themeId );
+
+	let themeTypeParam = themeType;
+	// Map theme tier values to the values expected by the signup flow.
+	const { themeTier } = getThemeTier( state, null, themeId );
+	// If there is no themeTier then there's nothing to map (dot-org themes don't have tiers)
+	if ( config.isEnabled( 'themes/tiers' ) && themeTier?.slug ) {
+		switch ( themeTier.slug ) {
+			case 'woocommerce':
+			case 'sensei':
+				themeTypeParam = BUNDLED_THEME;
+				break;
+			case 'partner':
+				themeTypeParam = MARKETPLACE_THEME;
+				break;
+			default:
+				themeTypeParam = themeTier.slug;
+				break;
+		}
+	}
+
 	const searchParams = {
 		ref: 'calypshowcase',
 		theme: themeId,
-		theme_type: themeType,
+		theme_type: themeTypeParam,
 		style_variation: styleVariationSlug,
-		intervalType: themeType === MARKETPLACE_THEME ? 'monthly' : 'yearly',
+		intervalType: themeTypeParam === MARKETPLACE_THEME ? 'monthly' : 'yearly',
 	};
 
 	// If the selected free theme belongs to the blog category, redirect users to the blog tailored flow
-	if ( themeType === FREE_THEME && tabFilter === 'blog' ) {
+	if ( themeTypeParam === FREE_THEME && tabFilter === 'blog' ) {
 		return addQueryArgs( `/setup/${ DESIGN_FIRST_FLOW }`, searchParams );
 	}
 
+	// Used to prefix the theme slug when creating `themeSlugWithRepo` which is then used in onboarding.
```

Comment:
I think passing premium here and reassembling the theme slug later is some kind of optimisation to save querying themes again but i'm a little surprised that that optimisation matters.

Some of the onboarding code related to this looked obsolete, but certainly not all of it. Figuring out how much each bit of code matters is not trivial (onboarding code is different fun to themes code). 

I think it's fine like this, i'm less certain about what happens if we move a theme in the  `premium/` dir to the personal tier but we're not worried about that right now. p1704451495018439-slack-C04DZ8M0GHW